mcpserver/tools.go

```diff
@@ -516,194 +506,204 @@ Background commands are unaffected by filesystem and any other kind of changes.
 				string(out), env.Config.Workdir, env.ID)), nil
 		}
 
+		// We want to update the repository whether the command fails or not
+		defer func() {
+			_ = repo.Update(ctx, env, "Run "+command, request.GetString("explanation", ""))
```

Comment:
🤔 i was expecting to see a lot more of this call given we've killed the propogateToXYZ method... is there only a few of them just because you've commented a lot of things out?